src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            [this[i], this[j]] = [this[j], this[i]];
+          }
+        } else {
+          if (String(this[i]) > String(this[j])) {
```

Comment:
When no compareFunction is provided, elements are sorted by converting them to strings and comparing their sequences of UTF-16 code units values. Your current implementation directly converts the elements to strings, which is correct, but keep in mind that this behavior should be explicitly mentioned in the documentation or comments for clarity.